tests/tools/onert_train/src/measure.h

```diff
@@ -47,42 +61,53 @@ class Measure
 
   void set(const int epoch, const int step)
   {
-    _results.clear();
-    _results.resize(epoch);
-    std::for_each(_results.begin(), _results.end(), [step](auto &v) { v.resize(step); });
+    _step_results.clear();
+    _step_results.resize(epoch);
+    std::for_each(_step_results.begin(), _step_results.end(), [step](auto &v) { v.resize(step); });
   }
 
   void run(const int epoch, const int step, const std::function<void()> &func)
   {
-    if (_results.empty() || _results.size() <= epoch || _results[epoch].size() <= step)
+    if (_step_results.empty() || _step_results.size() <= epoch ||
+        _step_results[epoch].size() <= step)
     {
       throw std::runtime_error("Please set the number of epochs and steps first");
     }
 
-    _results[epoch][step].time = nowMicros();
+    _step_results[epoch][step].time = nowMicros();
+
+    func();
+
+    _step_results[epoch][step].time = nowMicros() - _step_results[epoch][step].time;
```

Comment:
(Optional) To improve readability a bit:
```suggestion
    auto start_time = nowMicros();

    func();

    _step_results[epoch][step].time = nowMicros() - start_time;
```